lib/web/integrations_awsoidc.go

```diff
@@ -1070,6 +1070,18 @@ func (h *Handler) awsOIDCCreateAWSAppAccess(w http.ResponseWriter, r *http.Reque
 		return nil, trace.Wrap(err)
 	}
 
+	// If the integration name contains a dot, then the proxy must provide a certificate allowing *.<something>.<proxyPublicAddr>
+	if strings.Contains(integrationName, ".") {
+		// Teleport Cloud only provides certificates for *.<tenant>.teleport.sh, so this would generate an invalid address.
+		if h.GetClusterFeatures().Cloud {
+			return nil, trace.BadParameter(`Invalid integration name (%q) for enabling AWS Access. Please re-create the integration without the "."`, integrationName)
```

Comment:
Hrm, so how would this be backwards compatible in Cloud? If an integration with an invalid name already exists, what would happen? Can a user delete and recreate it, where would they see this error?

Reply:
This is not backwards compatible in Cloud.
Existing integration which have a `.` will not be usable for AWS Console access.

Users can delete the integration and create it again, but this time the checks will be applied and only valid integration names can be used.

They will see this error when they try to enable the AWS Console access using an integration as credentials source.
So, instead of seeing this message, they would see an error:
<img width="994" alt="image" src="https://github.com/user-attachments/assets/328f5d86-50a8-41e3-8b46-a14763e46657" />